radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/LedgerEntryStoreResult.java

```diff
@@ -17,31 +17,51 @@
 
 package com.radixdlt.store;
 
+import java.io.IOException;
 import java.util.Objects;
 
 public final class LedgerEntryStoreResult {
-	private static final LedgerEntryStoreResult SUCCESS = new LedgerEntryStoreResult(null);
+	private static final LedgerEntryStoreResult SUCCESS = new LedgerEntryStoreResult(null, null);
 
 	private final LedgerEntryConflict conflictInfo;
+	private final IOException exception;
 
-	private LedgerEntryStoreResult(LedgerEntryConflict conflictInfo) {
+	private LedgerEntryStoreResult(LedgerEntryConflict conflictInfo, IOException exception) {
```

Comment:
would it make sense to make it an ADT? something like:
```
interface LedgerEntryStoreResult
class ConflictResult(LedgerEntryConflict conflictInfo) extends LedgerEntryStoreResult
class ExceptionResult(IOException exception)  extends LedgerEntryStoreResult
```

Reply:
There are high chances that we might replace it with something simpler or already existing - `com.radix.atomos.Result` might be a good candidate (after some generalization and cleanup).